### ISSUE TITLE
Halide/C++ code failed to generate algorithms in  Windows CI over the weekend

### ISSUE THREAD

**Issue by antonysigma** [comp-imaging/ProxImaL#77]
The same revision https://github.com/comp-imaging/ProxImaL/commit/e0bbeba432cbc11d4a6d871f8c81f54a6b2ba776 passes the Windows CI on July 14, but the same code fails since July 19.

CI success on July 14: https://github.com/comp-imaging/ProxImaL/actions/runs/5564437916
CI failure today: https://github.com/antonysigma/ProxImaL/actions/runs/5557095504/job/15213363734

Error message: 
```
Run ninja -C proximal/halide/build ladmm-runtime
ninja: Entering directory `proximal/halide/build'
[1/6] Compiling C++ object src/user-problem/solver-generator.exe.p/.._.._subprojects_Halide-14.0.0-x86-64-windows_share_Halide_tools_GenGen.cpp.obj
cl : Command line warning D9002 : ignoring unknown option '-fno-rtti'
[2/6] Compiling C++ object src/user-problem/solver-generator.exe.p/linearized-admm-gen.cpp.obj
cl : Command line warning D9002 : ignoring unknown option '-fno-rtti'
[3/6] Linking target src/user-problem/solver-generator.exe
[4/6] Generating src/user-problem/ladmm_iter.[ah] with a custom command (wrapped by meson to set PATH, to set env)
FAILED: src/user-problem/ladmm_iter.lib src/user-problem/ladmm_iter.h 
"C:\hostedtoolcache\windows\Python\3.8.10\x64\Scripts\meson" "--internal" "exe" "--unpickle" "D:\a\ProxImaL\ProxImaL\proximal\halide\build\meson-private\meson_exe_solver-generator_13344a9f4d4ac864c6273976ca7dedc4382aeada.dat"
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\Scripts\meson.exe\__main__.py", line 7, in <module>
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\mesonbuild\mesonmain.py", line 294, in main
    return run(sys.argv[1:], launcher)
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\mesonbuild\mesonmain.py", line 282, in run
    return run_script_command(args[1], args[2:])
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\mesonbuild\mesonmain.py", line 223, in run_script_command
    return module.run(script_args)
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\mesonbuild\scripts\meson_exe.py", line 122, in run
    return run_exe(exe)
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\mesonbuild\scripts\meson_exe.py", line 76, in run_exe
    raise FileNotFoundError(p.returncode, strerror, cmd_args)
FileNotFoundError: [Errno 322122[57](https://github.com/antonysigma/ProxImaL/actions/runs/5557095504/job/15213363734#step:8:58)81] Failed to run due to missing DLLs, with path: D:\a\ProxImaL\ProxImaL\proximal\halide\subprojects\Halide-14.0.0-x86-64-windows\lib\Release;D:\a\ProxImaL\ProxImaL\proximal\halide\build\src/user-problem;D:/a/ProxImaL/ProxImaL/proximal/halide/subprojects/Halide-14.0.0-x86-64-windows/lib/Release: ['D:\\a\\ProxImaL\\ProxImaL\\proximal\\halide\\build\\src/user-problem\\solver-generator.exe', '-o', 'D:/a/ProxImaL/ProxImaL/proximal/halide/build/src/user-problem', '-g', 'ladmm_iter', '-e', 'static_library,h,stmt_html', 'machine_params=12,[62](https://github.com/antonysigma/ProxImaL/actions/runs/5557095504/job/15213363734#step:8:63)91000,40', 'target=host', 'auto_schedule=true', '-s', 'Mullapudi2016', '-p', 'autoschedule_mullapudi2016', 'n_iter=1', 'mu=0.333', 'lmb=3.0']
```

**Comment by antonysigma** [comment]
Meson version `v1.2.3` changed how `patch_directory` is handled in the wrap file handling source code compressed in zip. Reverting to meson `v1.2.0`...

**Comment by antonysigma** [comment]
Resolved in #102 . Closing.
